diff -dur SWIG-1.3.19.orig/configure.in SWIG-1.3.19/configure.in
--- SWIG-1.3.19.orig/configure.in	2004-01-07 10:57:19.000000000 +0100
+++ SWIG-1.3.19/configure.in	2004-01-07 10:58:02.384388040 +0100
@@ -954,7 +954,7 @@
 fi
 
 AC_MSG_CHECKING(for Ocaml header files)
-dirs="/usr/lib/ocaml/caml /usr/local/lib/ocaml/caml"
+dirs="/usr/include/caml /usr/lib/ocaml/caml /usr/local/lib/ocaml/caml"
 for i in $dirs; do
 	if test -r $i/mlvalues.h; then
 		AC_MSG_RESULT($i)
Only in SWIG-1.3.19: configure.in~
