--- SWIG-1.3.20/configure.in.orig	2003-12-09 19:53:52.000000000 +0100
+++ SWIG-1.3.20/configure.in	2003-12-19 00:45:55.436701752 +0100
@@ -1053,7 +1053,7 @@
 	dirs="/usr/include/php /usr/local/include/php /usr/include/php4 /usr/local/include/php4 /usr/local/apache/php"
 	for i in $dirs; do
 		echo $i
-		if test -r $i/php_config.h -o -r $i/php_version.h; then
+		if test -r $i/main/php_config.h -o -r $i/main/php_version.h; then
 			AC_MSG_RESULT($i)
 			PHP4EXT="$i"
 			PHP4INC="-I$PHP4EXT -I$PHP4EXT/Zend -I$PHP4EXT/main -I$PHP4EXT/TSRM"
