--- SWIG-1.3.19/Source/Modules/perl5.cxx.orig	Tue Dec  3 21:33:33 2002
+++ SWIG-1.3.19/Source/Modules/perl5.cxx	Sat Apr 12 10:56:55 2003
@@ -129,10 +129,10 @@
     for (i = 1; i < argc; i++) {
       if (argv[i]) {
 	if(strcmp(argv[i],"-package") == 0) {
-	  Printf(stderr,"*** -package is no longer supported\n*** use the directive '%module A::B::C' in your interface file instead\n*** see the Perl section in the manual for details.\n");
+	  Printf(stderr,"*** -package is no longer supported\n*** use the directive '%%module A::B::C' in your interface file instead\n*** see the Perl section in the manual for details.\n");
 	  SWIG_exit(EXIT_FAILURE);
 	} else if(strcmp(argv[i],"-interface") == 0) {
-	  Printf(stderr,"*** -interface is no longer supported\n*** use the directive '%module A::B::C' in your interface file instead\n*** see the Perl section in the manual for details.\n");
+	  Printf(stderr,"*** -interface is no longer supported\n*** use the directive '%%module A::B::C' in your interface file instead\n*** see the Perl section in the manual for details.\n");
 	  SWIG_exit(EXIT_FAILURE);
 	} else if (strcmp(argv[i],"-exportall") == 0) {
 	  export_all = 1;
