--- swig-1.3.30/Source/Modules/perl5.cxx.orig	2006-11-02 00:54:52.000000000 +0100
+++ swig-1.3.30/Source/Modules/perl5.cxx	2006-11-18 20:37:10.532401750 +0100
@@ -136,11 +136,11 @@
       if (argv[i]) {
 	if (strcmp(argv[i], "-package") == 0) {
 	  Printv(stderr,
-		 "*** -package is no longer supported\n*** use the directive '%module A::B::C' in your interface file instead\n*** see the Perl section in the manual for details.\n", NIL);
+		 "*** -package is no longer supported\n*** use the directive '%%module A::B::C' in your interface file instead\n*** see the Perl section in the manual for details.\n", NIL);
 	  SWIG_exit(EXIT_FAILURE);
 	} else if (strcmp(argv[i], "-interface") == 0) {
 	  Printv(stderr,
-		 "*** -interface is no longer supported\n*** use the directive '%module A::B::C' in your interface file instead\n*** see the Perl section in the manual for details.\n", NIL);
+		 "*** -interface is no longer supported\n*** use the directive '%%module A::B::C' in your interface file instead\n*** see the Perl section in the manual for details.\n", NIL);
 	  SWIG_exit(EXIT_FAILURE);
 	} else if (strcmp(argv[i], "-exportall") == 0) {
 	  export_all = 1;
