--- swig-4.1.1/Source/Modules/php.cxx.orig	2022-11-30 08:35:05.000000000 +0100
+++ swig-4.1.1/Source/Modules/php.cxx	2023-01-18 20:06:15.851111372 +0100
@@ -1532,12 +1532,16 @@ public:
 
     /* Insert cleanup code */
     for (i = 0, p = l; p; i++) {
+      if((i >= num_required) && (i < num_required+numopt))
+	Printf(cleanup,"\tif(arg_count > %d) {\n", i);
       if ((tm = Getattr(p, "tmap:freearg"))) {
 	Printv(cleanup, tm, "\n", NIL);
 	p = Getattr(p, "tmap:freearg:next");
       } else {
 	p = nextSibling(p);
       }
+      if ((i >= num_required) && (i < num_required+numopt))
+        Printf(cleanup,"}\n");
     }
 
     /* Insert argument output code */
