--- swig-2.0.2/Lib/guile/guile_scm_run.swg.orig	2010-02-28 00:26:02.000000000 +0100
+++ swig-2.0.2/Lib/guile/guile_scm_run.swg	2011-05-18 20:28:49.077624577 +0200
@@ -12,7 +12,11 @@
 extern "C" {
 #endif
 
+#if SCM_MAJOR_VERSION >= 2
+typedef scm_t_subr swig_guile_proc;
+#else
 typedef SCM (*swig_guile_proc)();
+#endif
 typedef SCM (*guile_destructor)(SCM);
 
 typedef struct swig_guile_clientdata {
--- swig-2.0.2/Lib/guile/guile_gh_run.swg.orig	2010-02-28 00:26:02.000000000 +0100
+++ swig-2.0.2/Lib/guile/guile_gh_run.swg	2011-05-18 20:28:35.984290806 +0200
@@ -15,7 +15,11 @@
 extern "C" {
 #endif
 
+#if SCM_MAJOR_VERSION >= 2
+typedef scm_t_subr swig_guile_proc;
+#else
 typedef SCM (*swig_guile_proc)();
+#endif
 
 #define SWIG_malloc(size) \
   SCM_MUST_MALLOC(size)
